### PR TITLE
Create TopBarNavigation Component

### PR DESCRIPTION
**Pull request by FacerOfGod**
Create a TopBar composable component for easy modularity.
- Leftside Icon: Easily toggle a back arrow icon for navigation.
- Middle: Ensures the title remains centered, regardless of additional icons.
- Rightside Icon: Customize with an optional clickable action icon.

### CODE REVIEW

**Comment by ohamz** [comment]
@FacerOfGod I get 100% line coverage on BottomNavigationMenu tests :)

File: app/src/androidTest/java/com/android/feedme/test/navigation/TopBarNavigationTest.kt [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+import com.kaspersky.kaspresso.testcases.api.testcase.TestCase
+import io.mockk.coVerify
+import io.mockk.every
+import io.mockk.mockk
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class TopBarNavigationTest : TestCase() {
```
Comment (JeannedeM) [inline app/src/androidTest/java/com/android/feedme/test/navigation/TopBarNavigationTest.kt:23]: Great and complete test. Nice Job !

**Review by JeannedeM — APPROVED** [review]
I just looked at your code, and everything looks good to me. Nice Job ! In particular i think it's a good practice that we should keep to define a new Color in the Color.kt file each time we use a specific custom color. You had a really great idea to do that. Also, good test coverage !

**Comment by xrsula** [comment]
Looks good! There's a small issue with the TopBar display, its display is not fixed size.
Screenshot 2024-04-11 at 16 48 31
I would recommend doing a height of 60.dp for the Box, and fix the fontWeight of the text to FontWeight(700) (:

**Comment by FacerOfGod** [comment]
Problem noted I will fix it

**Pull request merged**